### PR TITLE
add a flag to select target types to deregister instance from

### PR DESCRIPTION
**Pull request by yuxuanzhu0214**
**Changes**
Added a flag `deregister-target-types` which is a comma separated list of target types to deregister instance from. By default the value is `"classic-elb,target-group"`. Supported target types are `classic-elb` and `target-group`.

**Tests**
Manually applied the image in dev cluster. 

When `--deregister-target-types=classic-elb,target-group`, life cycle manager scan through all targets that match both target types.
```
time="2024-08-21T23:21:49Z" level=info msg="i-0ae3409fc1896603a> scanner starting"
time="2024-08-21T23:21:50Z" level=debug msg="cache hit => false, service => elasticloadbalancing.DescribeTargetGroups"
time="2024-08-21T23:21:50Z" level=debug msg="cache hit => false, service => elasticloadbalancing.DescribeLoadBalancers"
time="2024-08-21T23:21:50Z" level=info msg="i-0ae3409fc1896603a> checking targetgroup/elb membership"
time="2024-08-21T23:21:50Z" level=debug msg="len targetgroup: 42, len elb: 1"
```

When `- --deregister-target-types=classic-elb`, life cycle manager only scan through targets that are classic elb.
```
time="2024-08-21T23:08:44Z" level=info msg="i-03fffc250f0ffbe14> scanner starting"
time="2024-08-21T23:08:44Z" level=debug msg="cache hit => true, service => elasticloadbalancing.DescribeLoadBalancers"
time="2024-08-21T23:08:44Z" level=info msg="i-03fffc250f0ffbe14> checking targetgroup/elb membership"
time="2024-08-21T23:08:44Z" level=debug msg="len targetgroup: 0, len elb: 1"
```
Then life cycle manager deregisters the instance only from targets found from scanning.

### CODE REVIEW

**Review by ZihanJiang96 — APPROVED** [review]

**Review by tekenstam — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Build Stage
-FROM --platform=$BUILDPLATFORM golang:1.19-alpine as builder
+FROM --platform=$BUILDPLATFORM golang:1.21-alpine as builder
```
Comment (tekenstam) [inline Dockerfile:2]: If upgrading to go 1.21, there are a few other places that need to be changed. Please add the other changes needed for go 1.21, or remove this and make a separate PR for upgrading to go 1.21.

**Review by tekenstam — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (yuxuanzhu0214) [reply Dockerfile:2]: made a separate PR to bump golang version

**Review by shreyas-badiger — APPROVED** [review]

**Review by tekenstam — APPROVED** [review]
